### PR TITLE
Optimize `from_networkx()`

### PR DESCRIPTION
**Pull request by EulerPascal404**
Addresses issue #6964 by deleting unneeded copies by directly writing edges into a torch.Tensor rather than first converting to a list and then using torch.tensor() on top.

### CODE REVIEW

**Comment by EulerPascal404** [comment]
FYI, the unit test for this file seems to be broken(I tested it before I made any changes). The code changes I implemented didn't change the status of the test.

**Comment by EdisonLeeeee** [comment]
Thank you for the PR!

**Review by EdisonLeeeee — APPROVED** [review]

File: torch_geometric/utils/convert.py [hunk 1]
```diff
@@ -204,13 +204,10 @@ def from_networkx(
...
-    if isinstance(G, (nx.MultiGraph, nx.MultiDiGraph)):
-        edges = list(G.edges(keys=False))
-    else:
-        edges = list(G.edges)
-
-    edge_index = torch.tensor(edges, dtype=torch.long).t().contiguous()
+    edge_index = torch.empty(G.number_of_edges(), 2, dtype=torch.long)
+    for i, e in enumerate(G.edges):
+        edge_index[i, 0], edge_index[i, 1] = e
+    edge_index = edge_index.t().contiguous()
```
Comment (wsad1) [inline torch_geometric/utils/convert.py:210]: this is a for loop over edges, which can be slow if there are many edges. 
What if instead we call `del edges` after creating `edge_index` and allow the garbage collector to free up that memory if needed, this should make the OOM less likely.
```suggestion
    if isinstance(G, (nx.MultiGraph, nx.MultiDiGraph)):
        edges = list(G.edges(keys=False))
    else:
        edges = list(G.edges)

    edge_index = torch.tensor(edges, dtype=torch.long).t().contiguous()
    del edges
```

**Review by EulerPascal404 — COMMENTED** [review]
Sure, this is also a feasible solution - I just implemented the solution @rusty1s suggested in the issue. We should do this at other places in the file too, as per the initial issue.

**Review by rusty1s — APPROVED** [review]
Thanks! I changed it back to the original implementation of yours since we are iterating over all edges anyway here.

**Pull request merged**